latest/index.bs

```diff
@@ -331,24 +317,1002 @@ Conforming readers:
+in the container, its axes have `"type":"array"`, are unitless, and have default "name"s. The ith axis has `"name":"dim_i"`
+(these are the same default names used by [xarray](https://docs.xarray.dev/en/stable/user-guide/terminology.html)).
+<div class=example>
+For example, a 3D array at path `/my/data/array` defines the coordinate system:
+
+```json
+{
+    "name" : "/my/data/array",
+    "axes" : [
+        {"name": "dim_0", "type": "array"},
+        {"name": "dim_1", "type": "array"},
+        {"name": "dim_2", "type": "array"}
+    ]
+}
+```
+
+though this object should not and need not explicitly appear in metadata. 
+</div>
+
+
+The dimensionality of each array coordinate system equals the dimensionality of its corresponding zarr array.  The axis with
+name `"dim_i"` is the ith element of the `"axes"` list. The axes and their order align with the `shape`
+attribute in the zarr array attributes (in `.zarray`), and whose data depends on the byte order used to store
+chunks. As described in the [zarr array metadata](https://zarr.readthedocs.io/en/stable/spec/v2.html#arrays),
+the last dimension of an array in "C" order (row-major) are stored contiguously. For an array in "F"
+(column-major) order, the elements of the first dimension are stored contiguously.
+
+<div class=example>
+For example, if `/my/data/array/.zarray` contains:
+
+```json
+{
+    "chunks": [ 4, 3, 5 ],
+    "compressor": null,
+    "dtype": "|u1",
+    "fill_value": 0,
+    "filters": null,
+    "order": "C",
+    "shape": [ 4, 3, 5 ],
+    "zarr_format": 2
+}
+```
+
+Then `dim_0` has length 4, `dim_1` has length 3, and `dim_2` has length 5.
+</div>
+
+The name and axes names MAY be customized by including a `arrayCoordinateSystem` field in
+the `.zattr` metadata of the array whose value is a coordinate system object. The length of
+`axes` MUST be equal to the dimensionality. The value of `"type"` for each object in the 
+axes array MUST equal `"array"`.
+
+<div class=example>
+
+<pre class=include-code>
+path: examples/coordSystems/arrayCoordSys.json
+highlight: json
+</pre>
+
+</div>
+
+
+### Coordinate convention
+
+**The pixel/voxel center is the origin of the continuous coordinate system.**
+
+It is vital to consistently define relationship between the discrete/array and continuous/interpolated
+coordinate systems. A pixel/voxel is the continuous region (rectangle) that corresponds to a single sample
+in the discrete array, i.e., the area corresponding to nearest-neighbor (NN) interpolation of that sample.
+The center of a 2d pixel corresponding to the origin `(0,0)` in the discrete array is the origin of the continuous coordinate
+system `(0.0, 0.0)` (when the transformation is the identity). The continuous rectangle of the pixel is given by the
+half-open interval `[-0.5, 0.5) x [-0.5, 0.5)` (i.e., -0.5 is included, +0.5 is excluded). See chapter 4 and figure 4.1 of the ITK Software Guide [[itk]].
```

Comment:
I feel like this definition is a bit over-prescriptive? re http://alvyray.com/Memos/CG/Microsoft/6_pixel.pdf

In a sense, it feels to me like this paragraph is saying that the value at 0.25, 0.375 is the same as the value anywhere else in the pixel, which is clearly wrong and precludes any downstream interpolation...?

Reply:
This is to define the physical extent of the pixel, which is important for computing the physical bounds of the image, and for nearest neighbor interpolation. It does not preclude interpolation nor, when needed, extrapolation.

Reply:
Personally I think we should minimize unqualified use of terms like "pixel" and "voxel" in this spec for a few reasons:
- they imply a particular dimensionality (2D and 3D, respectively), but the spec should be N-dimensional.
- despite the Alvy Ray Smith essay, people still intuitively think of pixels and voxels as entities with area and volume. But as has been noted many times, "grid of square pixels / voxels" is just one of many valid resampling schemes for an image, and this spec should probably avoid promoting any particular resampling scheme. That's a concern for visualization software, not the storage layer.

I would propose re-writing the section about the origin of the coordinate system to say something like "...images stored in OME-NGFF are collections of samples. Each sample can be localized to a point in coordinate space via [reference to coordinate transformation metadata]. Resampling these images for purposes of visualization should obey the conventional mathematical rules for resampling. For example, when using nearest neighbor interpolation to generate a pixel or voxel representation of an image where each sample is localized to a unique point in some coordinate space, each pixel or voxel should be _centered_ around its respective point in coordinate space."

... or something like that.